Error when fetching HubDB table to existing dir

If you try to fetch a Hubdb table into an existing directory (like `hs hubdb fetch 1739916 folder`), you get the error:
```
[ERROR] Error: EISDIR: illegal operation on a directory, open '/Users/wspiro/DESKTOP_TEMP/local-cms-dev/folder'
    at Object.openSync (fs.js:440:3)
    at Object.writeFileSync (fs.js:1265:35)
    at downloadHubDbTable (/usr/local/lib/node_modules/@hubspot/cms-cli/node_modules/@hubspot/cms-lib/hubdb.js:166:12)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Command.<anonymous> (/usr/local/lib/node_modules/@hubspot/cms-cli/commands/hubdb.js:92:9) {
  errno: -21,
  syscall: 'open',
  code: 'EISDIR',
  path: '/Users/wspiro/DESKTOP_TEMP/local-cms-dev/folder'
```

--- a/packages/cms-cli/commands/hubdb.js
+++ b/packages/cms-cli/commands/hubdb.js
@@ -76,7 +76,7 @@ function configureHubDbFetchCommand(program) {
   program
     .version(version)
     .description('Fetch a HubDB table')
-    .arguments('<tableId> <dest>')
+    .arguments('<tableId> [dest]')
     .action(async (tableId, dest, command = {}) => {
       setLogLevel(command);
       logDebugInfo(command);
@@ -89,12 +89,9 @@ function configureHubDbFetchCommand(program) {
       }
       const portalId = getPortalId(command);
       try {
-        await downloadHubDbTable(
-          portalId,
-          tableId,
-          path.resolve(getCwd(), dest)
-        );
-        logger.log(`Downloaded HubDB table ${tableId} to ${dest}`);
+        const { filePath } = await downloadHubDbTable(portalId, tableId, dest);
+
+        logger.log(`Downloaded HubDB table ${tableId} to ${filePath}`);
       } catch (e) {
         logger.error(e);
       }

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -147,12 +147,12 @@ function convertToJSON(table, rows) {
 async function downloadHubDbTable(portalId, tableId, dest) {
   const table = await fetchTable(portalId, tableId);
 
+  dest = path.resolve(dest || `${table.name}.hubdb.json`);
+
   if (dest && fs.pathExistsSync(dest)) {
     validateJsonFile(dest);
-  } else if (dest) {
-    validateJsonPath(dest);
   } else {
-    dest = `${table.name}.hubdb.json`;
+    validateJsonPath(dest);
   }
 
   let totalRows = null;
@@ -176,6 +176,8 @@ async function downloadHubDbTable(portalId, tableId, dest) {
   });
 
   await fs.outputFile(dest, tableJson);
+
+  return { filePath: dest };
 }
 
 async function clearHubDbTableRows(portalId, tableId) {

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -152,7 +152,7 @@ async function downloadHubDbTable(portalId, tableId, dest) {
   } else if (dest) {
     validateJsonPath(dest);
   } else {
-    dest = table.name;
+    dest = `${table.name}.hubdb.json`;
   }
 
   let totalRows = null;

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -149,7 +149,7 @@ async function downloadHubDbTable(portalId, tableId, dest) {
 
   dest = path.resolve(dest || `${table.name}.hubdb.json`);
 
-  if (dest && fs.pathExistsSync(dest)) {
+  if (fs.pathExistsSync(dest)) {
     validateJsonFile(dest);
   } else {
     validateJsonPath(dest);

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -148,10 +148,8 @@ async function downloadHubDbTable(portalId, tableId, dest) {
   const table = await fetchTable(portalId, tableId);
 
   if (dest && fs.pathExistsSync(dest)) {
-    console.log('file');
     validateJsonFile(dest);
   } else if (dest) {
-    console.log('path', dest);
     validateJsonPath(dest);
   } else {
     dest = table.name;

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -11,6 +11,7 @@ const {
   publishTable,
   deleteRows,
 } = require('./api/hubdb');
+const { getCwd } = require('@hubspot/cms-lib/path');
 
 function validateJsonPath(src) {
   if (path.extname(src) !== '.json') {
@@ -147,7 +148,7 @@ function convertToJSON(table, rows) {
 async function downloadHubDbTable(portalId, tableId, dest) {
   const table = await fetchTable(portalId, tableId);
 
-  dest = path.resolve(dest || `${table.name}.hubdb.json`);
+  dest = path.resolve(getCwd(), dest || `${table.name}.hubdb.json`);
 
   if (fs.pathExistsSync(dest)) {
     validateJsonFile(dest);

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -148,8 +148,10 @@ async function downloadHubDbTable(portalId, tableId, dest) {
   const table = await fetchTable(portalId, tableId);
 
   if (dest && fs.pathExistsSync(dest)) {
+    console.log('file');
     validateJsonFile(dest);
   } else if (dest) {
+    console.log('path', dest);
     validateJsonPath(dest);
   } else {
     dest = table.name;
@@ -175,7 +177,7 @@ async function downloadHubDbTable(portalId, tableId, dest) {
     parser: 'json',
   });
 
-  await fs.writeFileSync(dest, tableJson);
+  await fs.outputFile(dest, tableJson);
 }
 
 async function clearHubDbTableRows(portalId, tableId) {

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -141,6 +141,10 @@ function convertToJSON(table, rows) {
 }
 
 async function downloadHubDbTable(portalId, tableId, dest) {
+  if (fs.pathExistsSync(dest)) {
+    validateJsonFile(dest);
+  }
+
   const table = await fetchTable(portalId, tableId);
 
   let totalRows = null;

--- a/packages/cms-lib/hubdb.js
+++ b/packages/cms-lib/hubdb.js
@@ -12,6 +12,12 @@ const {
   deleteRows,
 } = require('./api/hubdb');
 
+function validateJsonPath(src) {
+  if (path.extname(src) !== '.json') {
+    throw new Error('The HubDB table file must be a ".json" file');
+  }
+}
+
 function validateJsonFile(src) {
   try {
     const stats = fs.statSync(src);
@@ -22,9 +28,7 @@ function validateJsonFile(src) {
     throw new Error(`The "${src}" path is not a path to a file`);
   }
 
-  if (path.extname(src) !== '.json') {
-    throw new Error('The HubDB table file must be a ".json" file');
-  }
+  validateJsonPath(src);
 }
 
 async function addRowsToHubDbTable(portalId, tableId, rows, columns) {
@@ -141,11 +145,15 @@ function convertToJSON(table, rows) {
 }
 
 async function downloadHubDbTable(portalId, tableId, dest) {
-  if (fs.pathExistsSync(dest)) {
-    validateJsonFile(dest);
-  }
-
   const table = await fetchTable(portalId, tableId);
+
+  if (dest && fs.pathExistsSync(dest)) {
+    validateJsonFile(dest);
+  } else if (dest) {
+    validateJsonPath(dest);
+  } else {
+    dest = table.name;
+  }
 
   let totalRows = null;
   let rows = [];

